README.adoc

```diff
@@ -177,6 +177,36 @@ The `Makefile` provides commands for building and testing the code. Some depende
 
 Dependencies may be required to run some of the `Make` commands. Below are instructions on how to install them.
 
+== Creating mock for Interfaces
+This project is using teh dependency https://github.com/matryer/moq[moq]. Following the steps to use it.
+
+=== Install the required dependencies
+
+Run the following command.
+
+[source,shell]
+----
+$ go get -u github.com/matryer/moq
+----
+
+=== Creating mocks
+
+Execute the command the following command in the directory where is the interface.
+
+[source,shell]
+----
+$ moq -out <name_of_mock_file> . <name of interface>
```

Comment:
If we want to run this from the project root, this should show us how. Might I suggest a small change:

```sh
$ moq -out <name_of_mock_file> ./path/to/package <name of interface>
```

Reply:
The command needs to be executed where the Interface is. I am afraid that will not work.

Reply:
The following command worked for me:

```sh
$ moq -out ./pkg/web/apps/mocks_test.go ./pkg/web/apps AppService
```

Reply:
Cool to know. I will add it in the readme.